neuraxle/base.py

```diff
@@ -948,3 +1006,83 @@ class ResumableStepMixin:
     @abstractmethod
     def should_resume(self, data_container: DataContainer) -> bool:
         raise NotImplementedError()
+
+
+class Barrier(NonFittableMixin, NonTransformableMixin, BaseStep, ABC):
+    """
+    A Barrier step to be used in a minibatch sequential pipeline. It forces all the
+    data inputs to get to the barrier in a sub pipeline before going through to the next sub-pipeline.
+
+
+    ``p = MiniBatchSequentialPipeline([
+        SomeStep(),
+        SomeStep(),
+        Barrier(), # must be a concrete Barrier ex: Joiner()
+        SomeStep(),
+        SomeStep(),
+        Barrier(), # must be a concrete Barrier ex: Joiner()
+    ], batch_size=10)``
+
+    """
+
+    @abstractmethod
+    def join_transform(self, step: BaseStep, data_container: DataContainer) -> DataContainer:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def join_fit_transform(self, step: BaseStep, data_container: DataContainer) -> Tuple['Any', Iterable]:
+        raise NotImplementedError()
+
+
+class Joiner(Barrier):
+    """
+    A Special Barrier step that joins the transformed mini batches together with list.extend method.
+    """
+
+    def __init__(self, batch_size):
+        super().__init__()
+        self.batch_size = batch_size
+
+    def join_transform(self, step: BaseStep, data_container: DataContainer) -> Iterable:
+        """
+        Concatenate the pipeline transform output of each batch of self.batch_size together.
+
+        :param step: pipeline to transform on
+        :param data_container: data container to transform
+        :return:
+        """
+        data_container_batches = data_container.convolved_1d(
+            stride=self.batch_size,
+            kernel_size=self.batch_size
+        )
+
+        output_data_container = ListDataContainer.empty()
+        for data_container_batch in data_container_batches:
+            output_data_container.concat(
+                step._transform_core(data_container_batch)
+            )
```

Comment:
Please make the `DataContainer` raise an error on `concat` and rename it `BaseDataContainer`.

Reply:
it's a listdatacontainer resolving

Reply:
Sorry for the confusion.